mobius-core/src/main/java/com/spotify/mobius/MobiusLoop.java

```diff
@@ -184,12 +185,25 @@ public void dispose() {
 
   @Override
   public void dispose() {
-    synchronized (modelObservers) {
+    synchronized (disposeLock) {
```

Comment:
Better to just synchronize the entire method instead of having a specific lock. The synchronization on the `modelObservers` is still needed.